src/Perpetuum/Services/Channels/ChatCommands/AdminCommands.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Perpetuum.GenXY;
+using System.Collections.Generic;
+
+namespace Perpetuum.Services.Channels.ChatCommands
+{
+    /*
+     * Will contain all Chat Commands available to Admins
+     */
+    public class AdminCommands
+    {
+        public static string ZoneAddDecor(int? zoneId, int definition, int x, int y, int z, double qx, double qy, double qz, double qw, double scale, int cat)
+        {
+            Dictionary<string, object> dictionary = new Dictionary<string, object>()
+                {
+                    { "definition", definition },
+                    { "x", x*256 },
+                    { "y", y*256 },
+                    { "z", z*256 },
+                    { "quaternionX", qx },
+                    { "quaternionY", qy },
+                    { "quaternionZ", qz },
+                    { "quaternionW", qw },
+                    { "scale", scale },
+                    { "category", cat }
+                };
+
+            /* This foreach is unecessary in this particular case,
+             *  But we could allow for certain params to stay null (if they're optional) 
+             *  by splitting them into a required and a optional dictionary, validate the params
+             *  to then build and return the final result/command 
+             */
+            foreach (var entry in dictionary)
+            {
+                if (entry.Value == null)
+                {
+                    throw PerpetuumException.Create(ErrorCodes.RequiredArgumentIsNotSpecified);
+                }
+            }
```

Comment:
I think this is something like we want to see!
Somewhere we can specify required and optional args, their types, and quickly validate and cast them in from an ordered string array.

I don't think making a class for each command would be very maintainable. But I assume this is just for prototyping things out and I think there is some good opportunities to consolidate logic and make it more generic for any command's signature

Reply:
I was thinking that `AdminCommands.cs` will have a method for each command, so I would not build a new class for each but rather have them all in one class.

I imagine it would be the existing  `AdminCommandHandlers.cs` class which validates that the data-types are correct, and then it calls the new `AdminCommands` class to parse the correct data-types to the relevant method.

Then within `AdminCommands.zoneAddDecor()` (in this example), this method would then validate that the values of those provided data-types makes sense - i.e. it will contain the command-specific logic.

`AdminCommandHandlers` -> Validate Data Types, pass on to: `AdminCommands` -> Validate that params makes sense in given context (ex: teleport coords are within island)